challenge-13/challenge-13.js

```diff
@@ -129,22 +159,29 @@ objeto desse array, e adicione a frase abaixo na propriedade `estado`:
 - "[ESTADO] pertence ao Brasil."
 Atribua o novo array a uma variável chamada `map`.
 */
-// ?
+
+var map = newBrasil.map(function (item, index) {
+	return {id: index + 1, estado: item.estado + ' pertence ao brasil.'};
+});
 
 /*
 Mostre no console o array criado acima:
 */
-console.log( '\nnewBrasil agora com mais informações:' );
+console.log( '\nnewBrasil agora com mais informações:', map);
 // ?
 
 /*
 Filtre o array criado acima, retornando somente os estados que tiverem
 ID par. Atribua o valor à uma variável chamada `filter`.
 */
-// ?
-
+var filter = map.filter(function (item, index) {
+	if(( item.id ) % 2 === 0) {
+		return(item);
```

Comment:
-1
Não é necessário retornar o `item` quando você usa o `filter`. Basicamente o que acontece no `filter` é o seguinte:
Ele funciona como um `for`, passando por todos os itens do seu array. Nesse caso, o array é `map`.

Cada vez que ele passa por um item, ele avalia a sua instrução. Se a instrução for verdadeira, ele adiciona o item atual no array. Se for falsa, ele não adiciona. Simples assim.

Ou seja: se você fizer algo assim:

```js
var filter = map.filter(function(item, index) {
  return true;
});
```

Todos os itens do array `map` serão adicionados ao array em `filter`, pois todos serão avaliados como `true`. E não precisa ser valores `true` ou `false` exatamente. Todo valor é convertido para booleano antes de ser avaliado.

Por exemplo:

```js
var filter = map.filter(function(item, index) {
  return 2;
});
```

Simplesmente porque `2`, quando avaliado como booleano, retorna `true`. Lembra do `!!`? É mais ou menos isso que acontece: se você fizer `!!2`, o retorno é `true`.

Da mesma forma, para que você tenha um array vazio, é só retornar sempre false dentro do `.filter()`:

```js
var filter = map.filter(function(item, index) {
  return false;
});
```

Resumindo: você só precisa retornar a expressão! Se ela for avaliada como `true`, o item atual é adicionado ao array. Caso contrário, o item não é adicionado e o próximo item é testado, até que todos passem pelo filtro :)

Ficou mais claro agora? :)